### PR TITLE
expression: add NewDistAggFunc interface for mock tikv.

### PR DESCRIPTION
**Pull request by hanfei1991**
We add NewDistAggFunc function to reuse the aggregate function in mock-tikv. And we add `GetPartialResult` interface.

@shenli @coocood @zimulala PTAL

### CODE REVIEW

File: expression/aggregation.go [hunk 1]
```diff
@@ -332,6 +367,11 @@ func (sf *sumFunction) GetGroupResult(groupKey []byte) (d types.Datum) {
 	return sf.getContext(groupKey).Value
 }
 
+// GetPartialResult implements AggregationFunction interface.
+func (sf *sumFunction) GetPartialResult(groupKey []byte) []types.Datum {
+	return []types.Datum{sf.getContext(groupKey).Value}
```
Comment (shenli) [inline expression/aggregation.go]: Why not call GetGroupResult and convert its result to []Datum? So as the following functions.

Comment (hanfei1991) [inline expression/aggregation.go]: fixed

**Comment by shenli** [comment]
LGTM
@coocood @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**